Editor/AGS.Editor/Components/ViewsComponent.cs

```diff
@@ -61,7 +61,13 @@ protected override void ItemCommandClick(string controlID)
             {
                 View viewClicked = _items[_rightClickedID];
                 int oldNumber = viewClicked.ID;
-                int newNumber = Factory.GUIController.ShowChangeObjectIDDialog("View", oldNumber, 1, _items.Count);
+                // Note that the views are not sequential, there may be gaps in IDs
+                int maxNumber = 1;
+                foreach (var obj in _items)
+                {
+                    maxNumber = Math.Max(maxNumber, obj.Value.ID);
+                }
+                int newNumber = Factory.GUIController.ShowChangeObjectIDDialog("View", oldNumber, 1, maxNumber);

```

Comment:
This can be be reduced with LINQ

```suggestion
                int newNumber = Factory.GUIController.ShowChangeObjectIDDialog("View", oldNumber, 1, _items.Max(i => i.Value.ID));
```